test/e2e/sagemaker/tests/test_endpoint.py

```diff
@@ -39,91 +41,174 @@ def sagemaker_client():
 
 
 @pytest.fixture(scope="module")
-def single_variant_xgboost_endpoint():
-    endpoint_resource_name = random_suffix_name("single-variant-endpoint", 32)
-    config1_resource_name = endpoint_resource_name + "-config"
-    model_resource_name = config1_resource_name + "-model"
+def get_name_suffix():
+    return random_suffix_name("xgboost-endpoint", 32)
 
+
+@pytest.fixture(scope="module")
+def single_container_model(get_name_suffix):
+    model_resource_name = get_name_suffix + "-model"
     replacements = REPLACEMENT_VALUES.copy()
-    replacements["ENDPOINT_NAME"] = endpoint_resource_name
-    replacements["CONFIG_NAME"] = config1_resource_name
     replacements["MODEL_NAME"] = model_resource_name
 
-    model = load_resource_file(
-        SERVICE_NAME, "xgboost_model", additional_replacements=replacements
+    model_reference, model_spec = make_reference(
+        resource_prural=MODEL_RESOURCE_PLURAL,
+        resource_name=model_resource_name,
+        spec_file="xgboost_model",
+        replacements=replacements,
     )
-    logging.debug(model)
+    logging.debug(model_spec)
+    model_resource = make_resource(model_reference, model_spec)
+    assert model_resource is not None
+
+    yield (model_reference, model_resource)
+
+    k8s.delete_custom_resource(model_reference)
```

Comment:
Does this contain a conditional that checks existence?

Reply:
nope, this should not have been deleted by test